### PR TITLE
Support Fork PR to create package after merge

### PR DESCRIPTION
**Pull request by v-amolpatil**
Required items, please complete
   
   Change(s):
   - We need to support fork pr to create package after merge into master.

   Reason for Change(s):
   - Earlier we didnt had a support to create package for fork pr's. Now we are supporting fork pr to be able to generate package after merge into master.

   Version Updated:
   - NA

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

**Below are the screenshots while testing on private repo:**

1. **When PR is merged into master a new PR is generated as shown below where changes are related to Solution folder:**
image

2.  **Below is the newly generated PR and its content:**
image

3. **Once Auto generated PR is also merged into master it will not generate package again as shown in below screenshot:**
image

image

### CODE REVIEW

File: .github/workflows/package-on-fork.yaml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          $forkPRBranchName = "${{ env.BranchName }}"
+          Write-Host "Fork Branch Name is $forkPRBranchName"
+          $isAutomatedPR = $false
+          if ($forkPRBranchName -like '*automated-pr')
+          {
+            Write-Host "Skipping packaging as it is an automated fork pr!"
+            $isAutomatedPR = $true
+          }
+          Write-Output "isAutomatedPR=$isAutomatedPR" >> $env:GITHUB_OUTPUT
+          Write-Host "Is this Pull Request autogenerated $isAutomatedPR"
```
[the remark this answers is not in the document]
Comment (v-amolpatil) [reply .github/workflows/package-on-fork.yaml]: updated code

File: .github/workflows/package-on-fork.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+      - id: validateAndCreatePackage
+        name: validate-and-create-package
+        shell: pwsh
+        if: ${{ success() && steps.checkAutomatedPR.outputs.isAutomatedPR }}
```
Comment (Tichandr) [inline .github/workflows/package-on-fork.yaml:52]: It shouldn't be automated PR check. verify

File: .github/workflows/package-on-fork.yaml [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+          base: master
+          delete-branch: false
+          title: '[GitHub Bot] 🤖 Package Created For ${{ env.SOLUTION_NAME }} Solution'
+          body: |
+            Automation have successfully generated package for solution '${{ env.SOLUTION_NAME }}' based on Fork Pull request #${{ env.PULL_REQUEST_NUMBER }}.
+          labels: |
+            auto-package
+            Solution
+
+      - uses: fixpoint/azblob-upload-artifact@41eaaaa8fa50a5ba144d1d106c0e4ca11ca60c08
```
Comment (Tichandr) [inline .github/workflows/package-on-fork.yaml:118]: Please removed this. not required for now

File: .github/workflows/package-on-fork.yaml (hunk 2 above)
Comment (v-amolpatil) [reply .github/workflows/package-on-fork.yaml]: done

File: .github/workflows/package-on-fork.yaml (hunk 3 above)
Comment (v-amolpatil) [reply .github/workflows/package-on-fork.yaml]: done

**Review by mkchiliveri — APPROVED** [review]
Looks good to me

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**